SIL.Archiving/Generic/ArchivingFileSystem.cs

```diff
@@ -46,18 +47,15 @@ public static string CheckFolder(string folderName)
 				}
 				catch (UnauthorizedAccessException)
 				{
-					if (Platform.IsLinux)
+					if (Platform.IsUnix && (folderName.StartsWith("/var/lib/SIL") || folderName.StartsWith("/Users/Shared")))
 					{
-						if (folderName.StartsWith("/var/lib/SIL"))
-						{
-							// by default /var/lib isn't writable on Linux, so we can't create a new
-							// directory. Create a folder in the user's home directory instead.
-							var endFolder = folderName.Substring("/var/lib/SIL".Length).TrimStart('/');
-							folderName = Path.Combine(
-								Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
-								"SIL", endFolder);
-							return CheckFolder(folderName);
-						}
+						// If the default directory isn't writable on a Unix system (we couldn't create the SIL directory)
+						// fall back to creating a folder in the users home directory instead.
+						var endFolder = folderName.Substring("/var/lib/SIL".Length).TrimStart('/');
```

Comment:
Doesn't `"/var/lib/SIL".Length` need to be different in the `Platform.IsMac` case?